atmat/lattice/atparamscan.m

```diff
@@ -0,0 +1,275 @@
+function [parmelem,pvals] = atparamscan(ring,parmelem,varargin)
+%ATPARAMSCAN    Private function. Updates the RingParam element
+%
+%NEWPARMS=ATPARAMSCAN(RING,PARMS,VARARGIN)
+%
+%  See also ATGETRINGPROPERTIES, ATSETRINGPROPERTIES
+
+global GLOBVAL %#ok<GVMIS>
+
+TWO_PI_ERROR = 1.e-4;
+
+store=struct();
+
+pvals=cellfun(@getany, varargin, 'UniformOutput', false);
+
+    function v=getany(param)
+        switch lower(param)
+            case {'famname','name'}
+                v=getname(ring);
+            case 'particle'
+                v=getparticle(ring);
+            case 'energy'
+                v=getenergy(ring);
+            case 'periodicity'
+                v=getnp(ring);
+            case 'cell_harmnumber'
+                v=getcell_h(ring);
+            case 'cavpts'
+                v=getmaincav(ring);
+            case 'beta'
+                v=get_beta(ring);
+            case 'gamma'
+                v=get_gamma(ring);
+            case 'cell_length'
+                v=get_cell_length(ring);
+            case {'harmnumber','harmonic_number'}
+                v=getnp(ring)*getcell_h(ring);
+            case 'circumference'
+                v=getnp(ring)*get_cell_length(ring);
+            case 'rf_frequency'
+                v=get_unique(ring(getmaincav(ring)), 'Frequency');
+            case 'cell_rf_voltage'
+                v=get_sum(ring(getmaincav(ring)), 'Voltage');
+            case 'rf_voltage'
+                v=getnp(ring) * get_sum(ring(getmaincav(ring)), 'Voltage');
+            case 'rf_timelag'
+                v=get_unique(ring(getmaincav(ring)), 'TimeLag');
+            case 'cell_revolution_frequency'
+                v=get_cell_frev(ring);
+            case 'revolution_frequency'
+                v=get_cell_frev(ring) / getnp(ring);
+            case 'brho'
+                clight=PhysConstant.speed_of_light_in_vacuum.value;
+                energy=getenergy(ring);
+                rest_energy=getparticle(ring).rest_energy;
```

Comment:
@lfarv
Indexing into Function Call Results is only working for R2022a++
this nice feature makes an error for earlier version.

Reply:
same on lines #217 and 228